drafts/signals-everywhere.md

```diff
@@ -0,0 +1,162 @@
+---
+title: Signals, Signals everywhere
+subtitle: "Or: dawn of a new era as web frameworks converge"
+domain: software-engineering-corner.zuehlke.com
+tags: signals
+cover:
+publishAs: culas
+hideFromHashnodeCommunity: false
+saveAsDraft: true
+---
+
+After decades of JavaScript Wild West exploration with countless frameworks and libraries left in the dust, the recent emergence of Signals is the best thing to happen since ES5.
+If you're unsure about what Signals are and why they're important, it's time to learn.
+**Every web developer should know and understand the concept of Signals** to build interactive experiences on the web going forward.
+
+### The Problem
+
+All JavaScript frameworks essentially try to solve one single challenge at their core: **"keeping the UI in sync with the state"** (see [The deepest reason why modern JavaScript frameworks exist](https://medium.com/dailyjs/the-deepest-reason-why-modern-javascript-frameworks-exist-933b86ebc445)).
+One way or another, we need to update our visual interface every time the state changes, something React describes as `UI = fn(state)`.
+
+Vanilla JavaScript only gives us rudimentary, imperative APIs to achieve that.
+The complexity of an interactive website rapidly grows beyond what is reasonably maintainable without the use of third-party libraries and frameworks.
+They can provide structured ways to declaratively define our views, manage state and leave the synching itself to the frameworks.
+Each new framework proposed a new or slightly different way to make our UI reactive, while still providing good developer ergonomics (or "developer experience", DX).
+Angular uses a conservative change detection algorithm, while React introduced a virtual DOM for better performance.
+Svelte went yet another way and used a compiler to analyse dependencies at build time.
+
+All these approaches have their own advantages and drawbacks.
+In any case, learning a new framework meant learning completely new concepts.
+In-depth Angular knowledge is only vaguely transferrable to writing React applications.
+
+Signals could change this forever and have already done so.
+
+### The Conquest of Signals
+
+Solid, released in 2021, combined the best of different frameworks.
+It leverages the power of a compiler like Svelte.
+They've adopted React's philosophy of unidirectional data flow and composition.
+But for one significant feature, Solid's creator Ryan Carniato went further back in time for inspiration: Knockout.
+
+Released back in 2010, Knockout presented a reactivity concept they called "observables".
+They are described as **"special JavaScript objects that can notify subscribers about changes, and can automatically detect dependencies"** in the [documentation](https://knockoutjs.com/documentation/observables.html).
```

Comment:
I would probably remove the second part of the sentence "in the documentation" and link the quote directly or add (documentation) in brackets after the quote.